.travis.yml

```diff
@@ -16,8 +16,15 @@ jobs:
   include:
     - stage: Tests
       name: Unit Tests
-      script: mvn clean test --batch-mode
-      after_success: mvn -DrepoToken=$COVERALLS_REPO_TOKEN org.jacoco:jacoco-maven-plugin:report org.eluder.coveralls:coveralls-maven-plugin:report
+      before_script:
+        - curl -L https://codeclimate.com/downloads/test-reporter/test-reporter-latest-linux-amd64 > ./cc-test-reporter
+        - chmod +x ./cc-test-reporter
+        - ./cc-test-reporter before-build
+      script: mvn package --batch-mode
+      after_success: mvn -DrepoToken=$COVERALLS_REPO_TOKEN org.eluder.coveralls:coveralls-maven-plugin:report
+      after_script:
```

Comment:
Is the `after_script` by intention? To my understanding that phase is executed also in case of a failure. Are we interested in the results also in case of an earlier failure? Eg the coverall report is uploaded only in case of a success. Looks stange to me.

Reply:
This is the ways it's [documented for travis](https://docs.codeclimate.com/v1.0/docs/travis-ci-test-coverage).

Reply:
I guess the coverage results are also valid if tests are failing, right?

Reply:
... philosophical question. But why handling coveralls and codeclimate in a different way?

Reply:
joined not in `after_script`